### PR TITLE
Add AssertionsArgumentOrder recipe for RSPEC-3415. (issue #275)

### PR DESCRIPTION
**Pull request by pway99**

### CODE REVIEW

**Comment by pway99** [comment]
@timtebeek,  if you have a minute and can have a look at this one, that would be great.  Thanks!

File: src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        private final MethodMatcher[] testNgMatcher = new MethodMatcher[] {
+                new MethodMatcher("org.testng.Assert assertSame(..)"),
+                new MethodMatcher("org.testng.Assert assertNotSame(..)"),
+                new MethodMatcher("org.testng.Assert assertEquals(..)"),
+                new MethodMatcher("org.testng.Assert assertNotEquals(..)")
+        };
+
+        private final MethodMatcher[] newListMatchers = new MethodMatcher[]{
+                new MethodMatcher("javajava.util.List of(..)"),
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java:69]:
```suggestion
                new MethodMatcher("java.util.List of(..)"),
```

**Comment by timtebeek** [comment]
Looks good to me! Haven't run it yet, as I'm just heading out to the office. Might want to run this one against Wicket and see if the tests still pass after that. With a large code base there's likely going to be some instances swapped, as well as edge cases we might not have covered yet.

File: src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java:16]: Slightly puzzled by the `assertj` package. Perhaps move to `testing` above.

**Comment by timtebeek** [comment]
Ran this on Apache Wicket, and noticed a weird failure after swapping arguments:
```
mvn org.openrewrite.maven:rewrite-maven-plugin:4.36.0:run \
  -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-testing-frameworks:1.31.0-SNAPSHOT \
  -DactiveRecipes=org.openrewrite.java.testing.assertj.AssertionsArgumentOrder
```
```
[ERROR] Failures: 
[ERROR]   AppendingStringBufferTest.equalsToCharSequence:70
  expected: java.lang.String@52cb4f50<123456789>
  but was: org.apache.wicket.util.string.AppendingStringBuffer@25a5c7db<123456789>
```
```diff
diff --git a/wicket-util/src/test/java/org/apache/wicket/util/string/AppendingStringBufferTest.java b/wicket-util/src/test/java/org/apache/wicket/util/string/AppendingStringBufferTest.java
index 7eedb46fed..e1ac4b0883 100644
--- a/wicket-util/src/test/java/org/apache/wicket/util/string/AppendingStringBufferTest.java
+++ b/wicket-util/src/test/java/org/apache/wicket/util/string/AppendingStringBufferTest.java
@@ -67,7 +67,7 @@ public class AppendingStringBufferTest
                AppendingStringBuffer asb = new AppendingStringBuffer("123456789");
                StringBuilder sb = new StringBuilder("123456789");
                assertEquals(asb, sb);
-               assertEquals(asb, "123456789");
+               assertEquals("123456789", asb);
 
                sb = new StringBuilder("01234567890");
                assertNotEquals(asb, sb);
```
Could just be a strange case related to the `org.apache.wicket.util.string.AppendingStringBuffer`, but might affect others as well.

**Comment by timtebeek** [comment]
All other changes looked ok: https://github.com/apache/wicket/pull/546 :tada:

**Comment by timtebeek** [comment]
Might also want to reference this recipe from [META-INF/rewrite/junit5.yml#L181](https://github.com/openrewrite/rewrite-testing-frameworks/blob/assertions-argument-order/src/main/resources/META-INF/rewrite/junit5.yml#L181)

**Comment by timtebeek** [comment]
Any particular reason you didn't include JUnit 5 Assertions `assertNotEquals`?

File: src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java (hunk 1 above)
Comment (pway99) [reply src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java:69]: nice catch thanks!

File: src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java (hunk 2 above)
Comment (pway99) [reply src/main/java/org/openrewrite/java/testing/assertj/AssertionsArgumentOrder.java:16]: another good catch thanks!
